modules/social_features/social_topic/src/Plugin/GraphQL/QueryHelper/TopicQueryByTypeHelper.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Drupal\social_topic\Plugin\GraphQL\QueryHelper;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\Query\QueryInterface;
+use Drupal\graphql\GraphQL\Buffers\EntityBuffer;
+use Drupal\node\Entity\Node;
+use Drupal\social_graphql\GraphQL\ConnectionQueryHelperBase;
+use Drupal\social_graphql\Wrappers\Cursor;
+use Drupal\social_graphql\Wrappers\Edge;
+use GraphQL\Deferred;
+use GraphQL\Executor\Promise\Adapter\SyncPromise;
+
+/**
+ * Loads topics.
+ */
+class TopicQueryByTypeHelper extends ConnectionQueryHelperBase {
+
+  /**
+   * The topic type id.
+   *
+   * @var string
+   */
+  protected string $type;
+
+  /**
+   * TopicQueryHelper constructor.
+   *
+   * @param string $sort_key
+   *   The key that is used for sorting.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The Drupal entity type manager.
+   * @param \Drupal\graphql\GraphQL\Buffers\EntityBuffer $graphql_entity_buffer
+   *   The GraphQL entity buffer.
+   * @param string $type
+   *   The topic type.
+   */
+  public function __construct(string $sort_key, EntityTypeManagerInterface $entity_type_manager, EntityBuffer $graphql_entity_buffer, string $type) {
+    parent::__construct($sort_key, $entity_type_manager, $graphql_entity_buffer);
+    $this->type = $type;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuery(): QueryInterface {
+    $term_ids = $this->entityTypeManager->getStorage('taxonomy_term')
+      ->getQuery()
+      ->condition('uuid', $this->type)
+      ->execute();
+
+    return $this->entityTypeManager->getStorage('node')
+      ->getQuery()
+      ->currentRevision()
+      ->accessCheck(TRUE)
+      ->condition('type', 'topic')
+      ->condition('field_topic_type', $term_ids ?: [0], 'IN');
```

Comment:
The following should work as a single query :)

```suggestion
    return $this->entityTypeManager->getStorage('node')
      ->getQuery()
      ->currentRevision()
      ->accessCheck(TRUE)
      ->condition('type', 'topic')
      ->condition('field_topic_type.0.entity:taxonomy_term.uuid', $this->type);
```

For other places where you may need to load an entity by `uuid` you can do so with `$this->entityRepository->loadEntityByUuid('entity_type', $uuid)` :D That avoids scenarios in which the `uuid` is not actually in the `uuid` field (which I haven't seen done, but is technically possible 🙈 )

Reply:
Yeah, good to know 😁